### ISSUE TITLE
Is there a way to disable <c-w> for Firefox tab deletion and use it for word deletion, instead?

### ISSUE THREAD

**Issue by Lovewyrm** [tridactyl/tridactyl#1428]
Long term pentadactyl usage, as well as regular *nix usage has burned it into my muscles, and it is actually one of the biggest things I miss.

**Comment by glacambre** [comment]
Yes, but you have to recompile firefox with [this patch](https://github.com/glacambre/firefox-patches). It can also be done by patching your omni.ja file but I don't have the specific steps in mind for now.

I've been toying with the idea of making omni.ja patching easy with a single command but I haven't implemented anything yet.

**Comment by aplaice** [comment]
Another simpler, but slightly less "effective" alternative, is to mess around with `ui.key.accelKey` in `about:config`. If you change it from `17` (the default — `Ctrl`) to `18` (`Alt`) then the shortcuts for `<C-w>`, `<C-t>`, `<C-n>` etc. will now be `<A-w>`, `<A-t>`, `<A-n>`. These shortcuts will still be "reserved", but if you care more about having `Ctrl` keybindings free, then you're "ahead". (OTOH remapping virtually all the browser shortcuts, might mess with muscle memory.) Another alternative value for `ui.key.accelKey` is `91` which is the "Window key" (`DOM_VK_WIN`) and both "super" and "hyper" on X11. (This might collide with your WM keybindings...)

Regarding patching `omni.ja`, the last time that I tried, the process described in [these](https://shallowsky.com/blog/tech/web/modifying-omni.ja.html) [articles](https://shallowsky.com/blog/tech/web/script-omni.ja.html) worked. ([Direct link to the patching script, under GPLv2](https://github.com/akkana/scripts/blob/master/kitfox-patch).)

**Comment by bovine3dom** [comment]
Related - #1590.

**Comment by arturakb** [comment]
> Another simpler, but slightly less "effective" alternative, is to mess around with `ui.key.accelKey` in `about:config`. If you change it from `17` (the default — `Ctrl`) to `18` (`Alt`) then the shortcuts for `<C-w>`, `<C-t>`, `<C-n>` etc. will now be `<A-w>`, `<A-t>`, `<A-n>`. These shortcuts will still be "reserved", but if you care more about having `Ctrl` keybindings free, then you're "ahead". (OTOH remapping virtually all the browser shortcuts, might mess with muscle memory.) Another alternative value for `ui.key.accelKey` is `91` which is the "Window key" (`DOM_VK_WIN`) and both "super" and "hyper" on X11. (This might collide with your WM keybindings...)
> 
> Regarding patching `omni.ja`, the last time that I tried, the process described in [these](https://shallowsky.com/blog/tech/web/modifying-omni.ja.html) [articles](https://shallowsky.com/blog/tech/web/script-omni.ja.html) worked. ([Direct link to the patching script, under GPLv2](https://github.com/akkana/scripts/blob/master/kitfox-patch).)

I was thinking to give up on Firefox because of the <C-w> binding, the number of times I've pressed it to delete a word and closed the tab had started giving anxiety issues. This answer should be on Firefox forum about the <C-w>. Thank you so much!

**Comment by WloHu** [comment]
@aplaice Works great. I was worried that it may also remap `<C-PageUp>` `<C-PageDn>` which allow me to get away from Firefox "protected" tabs.

**Comment by bovine3dom** [comment]
glacambre has a patch for Firefox on Linux that allows you to bind to these keys with installation instructions here: https://github.com/glacambre/firefox-patches/issues/1

**Comment by ne9z** [comment]
There is a way to disable <c-w> for Firefox tab deletion and use it for word deletion, instead.  The method is to use GTK Key Theme feature.

With vanilla Firefox.  Use dconf-editor, navigate to org/gnome/desktop/interface, change `gtk-key-theme` option to the String `"Emacs"`.  The change is effective immediately without restarting Firefox.  More enabled keys with Emacs theme:
https://www.gnu.org/software/bash/manual/html_node/Readline-Bare-Essentials.html

Examples: 

C-b

    Move back one character. 
C-f

    Move forward one character. 
DEL or Backspace

    Delete the character to the left of the cursor. 
C-d

    Delete the character underneath the cursor. 
C-a

    Move to the start of the line. 
C-e

    Move to the end of the line. 
M-f

    Move forward a word, where a word is composed of letters and digits. 
M-b

    Move backward a word. 

C-k

    Kill the text from the current cursor position to the end of the line.
M-d

    Kill from the cursor to the end of the current word, or, if between words, to the end of the next word. Word boundaries are the same as those used by M-f.
M-DEL

    Kill from the cursor to the start of the current word, or, if between words, to the start of the previous word. Word boundaries are the same as those used by M-b.
C-w

    Kill from the cursor to the previous whitespace. This is different than M-DEL because the word boundaries differ.

In Emacs parlance, C- prefix is for Control key. M- prefix is for Alt key.

**Comment by ne9z** [comment]
Note that all the key combinations for line editing only works, when the cursor is in a text input field.  Outside they would still follow Firefox defaults, i.e. C-w will still close the tab.

**Comment by Mikilio** [comment]
The answer to this question is [autoconfig.js](https://support.mozilla.org/en-US/questions/1378404#answer-1510431)
